### ISSUE TITLE
Default fetch polyfill breaks ky-universal

### ISSUE THREAD

**Issue by paambaati** [zeit/next.js#12761]
# Bug report

I use [`ky-universal`](https://github.com/sindresorhus/ky-universal), which is built on top of `node-fetch`. After upgrading to Next.js 9.4, it has started throwing `"Only absolute URLs are supported"` errors on the Node.js side, even though an absolute URL is set and it works on Next.js 9.3.x.

## To Reproduce

1. Use [`ky-universal`](https://github.com/sindresorhus/ky-universal) in a Next.js project to make a request on the server side.
2. Watch it throw a `"Only absolute URLs are supported"` error.

## Expected behavior

Shouldn't throw an error, and should work like it did in Next.js 9.3.x.

## System information

- OS: macOS
- Version of Next.js: 9.4.0
- Version of Node.js: 14.1.0

## Additional context

It looks like the default polyfill is interfering with the library somehow, but I can't figure out what/how. Also, pasting from https://github.com/zeit/next.js/pull/12353#issuecomment-627064091 —

> `node-fetch` has an upcoming v3 release (in beta now) that fixes a [long-standing issue](https://github.com/node-fetch/node-fetch/issues/386), and [`ky-universal` is in the process of using it in their next release](https://github.com/sindresorhus/ky-universal/pull/19), which is vital when working with large API payloads.

Similar reports —

1. https://github.com/zeit/next.js/pull/12353#issuecomment-625114086
2. https://github.com/zeit/next.js/discussions/12732

**Comment by stfnsr** [comment]
I thought that with Next.js 9.4 it would be possible to use ky instead of ky-universal, as fetch is already polyfilled by Next.js.
Is this not the case?

**Comment by joshwaiam** [comment]
The suggestion to use ky instead of ky-universal is something I did not try and I thought would work.  However, it looks like the ky-universal package is more than simply a polyfill.  When testing in my project by using just the ky package instead of ky-universal, I get a whole ton of other errors.

**Comment by stfnsr** [comment]
Yep, can confirm.
Same issue with "Unexpected token 'export'" :/
Will stick to vanilla fetch for the time being.

**Comment by YukiKitagata** [comment]
I have encountered the same error.

For the time being, I added the following code and the error is gone.

```
if (typeof window === 'undefined')
  (global as any).fetch = require('node-fetch');
```

**Comment by paambaati** [comment]
@YukiKitagata Thanks for the temporary workaround. Where exactly is this used? `_app`?

**Comment by YukiKitagata** [comment]
@paambaati I'm using ky.extensions, so I'm adding it before call ky.extensions.

**Comment by paambaati** [comment]
@YukiKitagata Can you show us an example? I can't seem to find this extensions API in [Ky's documentation](https://github.com/sindresorhus/ky#readme).

**Comment by YukiKitagata** [comment]
@paambaati 

Sorry, that was a typo on my part.
ky.extensions is a mistake for ky.extensions.

I tried adding it to the top of _app and it worked well.

**Comment by joshwaiam** [comment]
I was able to put the fix by @YukiKitagata into practice.  Wanted to paste it here for @paambaati and anyone else that may be unclear of how to implement it.  I installed node-fetch to my project.  Then in my _app.tsx, at the very top, I added the code block and ky works as expected.

```
import ky from 'ky-universal';

if (typeof window === 'undefined')
  (global as any).fetch = require('node-fetch');

function MyApp({ Component, pageProps }: AppProps): JSX.Element {
```

**Comment by timneutkens** [comment]
Hey! Thanks for the report, I've fixed the issue here: #12804

**Comment by paambaati** [comment]
@timneutkens
![](https://media.tenor.com/images/c773ebfe4991507a6968643329de6fd9/tenor.gif)

**Comment by paambaati** [comment]
@timneutkens Thanks, I can confirm that this is fixed and `ky-universal` works fine on `9.4.1-canary.1`.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
